better handling of RFB 3.3

Mauriceter opened the pull request:
## Description

Fix for issue: https://github.com/Pennyw0rth/NetExec/issues/942, when error on connection is not about security types not being supported, it tries RFB 3.3  to check if it is a VNC service and the security types used.

I also updated the printed host info to indicate if security type 1 (no password) is supported.

## Type of change
Insert an "x" inside the brackets for relevant items (do not delete options)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Deprecation of feature or functionality
- [ ] This change requires a documentation update
- [ ] This requires a third party update (such as Impacket, Dploot, lsassy, etc)

## Setup guide for the review
Please provide guidance on what setup is needed to test the introduced changes, such as your locally running machine Python version & OS, as well as the target(s) you tested against, including software versions.
In particular:
- Bug Fix: Please provide a short description on how to trigger the bug, to make the bug reproducable for the reviewer.
- Added Feature/Enhancement: Please specify what setup is needed in order to test the changes. E.g. is additional software needed? GPO changes required? Specific registry settings that need to be changed?

## Screenshots (if appropriate):
Before:

<img width="724" height="375" alt="image" src="https://github.com/user-attachments/assets/2d26d8b6-5e2d-40db-bd44-b156dda7902c" />

After:

<img width="773" height="478" alt="image" src="https://github.com/user-attachments/assets/c10f7734-21eb-47eb-8cd8-257a6484980c" />


## Checklist:
Insert an "x" inside the brackets for completed and relevant items (do not delete options)

- [ ] I have ran Ruff against my changes (via poetry: `poetry run python -m ruff check . --preview`, use `--fix` to automatically fix what it can)
- [ ] I have added or updated the `tests/e2e_commands.txt` file if necessary (new modules or features are _required_ to be added to the e2e tests)
- [ ] New and existing e2e tests pass locally with my changes
- [ ] If reliant on changes of third party dependencies, such as Impacket, dploot, lsassy, etc, I have linked the relevant PRs in those projects
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (PR here: https://github.com/Pennyw0rth/NetExec-Wiki)

NeffIsBack commented:
Thanks for the bug fix!

Any advice on the setup? Do you know how i can configure a machine to test that PR?

NeffIsBack commented:
Thanks for the explanation! I'll take a deeper look at this soon so i can comment on it properly.

NeffIsBack reviewed (changes requested):
For reasons it hangs on my side indefinitely as well if i have no auth set in vnc. Unfortunately i don't have the time at the moment to check why the default timeout (1s) is not working.

I removed the manual proto flow so we can use the `enum_host_info()` function. I think we should probe the host in there and then display the info accordingly. Otherwise this looks good so far.

Mauriceter commented:
I moved the probing to enum_host_info and managed to fix the hanging by using a security type 2 instead of 1 with the default connection.
<img width="848" height="276" alt="image" src="https://github.com/user-attachments/assets/2a9fa518-9056-4d66-9de3-1dea098c66a1" />

Mauriceter commented:
renamed sec to stype, it is also displayed in the host infos.
<img width="948" height="345" alt="image" src="https://github.com/user-attachments/assets/b82ef40c-57ab-49c3-95a1-d8badd83a975" />

NeffIsBack commented:
Great, but isn't that the same information as "No Auth: bool"?

Mauriceter commented:
Yeah I agree, I think I will put the supported security types in an `nxc_logger.info()` so the information can still be accessed if needed.

If Im not wrong, the  RFC only defines 1 and 2. Other security types are vendor/software specifics.

Mauriceter commented:
Removed security types from print_host_info(). Also corrected the false assumption that version was ether 3.3 or 3.8 and changed self.RFBversion type to float instead or str.

NeffIsBack commented:
> Im not sure what you mean by "why the RFB verison should matter", depending on the version the protocol isn't the same, before this PR when only version 3.3 is supported, it would just ignore the the service as if it doesn't exist. Alos the parsing is different to check if null auth is allowed.

I thought by specifying `-rfbversion 3.x` the x11vnc server would just pin the server to that version, hence auth would work even with 3.3 and 3.7. But as you said, the server still supports 3.8 and asyauth will use 3.8 to connect, so never mind.

NeffIsBack commented:
> Okay so remove the unsupported flag and test auth anyway right?

Yeah i think that should be good for now.
